pkg/controller/daemon/daemoncontroller.go

```diff
@@ -659,6 +678,14 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *extensions.DaemonSet)
 				sort.Sort(podByCreationTimestamp(daemonPods))
 				if v1.IsPodReady(daemonPods[0]) {
 					numberReady++
+					ready = true
+				}
+				if util.IsPodUpdated(dsPodTemplateSpecHash, daemonPods[0]) {
+					updatedNumberScheduled++
+					updated = true
+				}
+				if ready && updated {
+					numberAvailable++
```

Comment:
This is not what available should be. Available is the # of pods (old or new) that are ready for at least minReadySeconds.